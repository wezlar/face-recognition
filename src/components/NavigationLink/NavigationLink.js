/** @jsx jsx */
import { jsx, css } from '@emotion/core';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const style = css`
  background-color: #3939a0;
  display: block;
  padding: 20px 30px;
  color: #fff;
  &:hover {
    background-color: #272781;
  }
`;

export const NavigationLink = (props = {}) => {
  const { isActive = false, label, url = '/' } = props;
  
  return (
    <li>
      <Link to={url} css={style}>
        {label}
      </Link>
    </li>
  );
};

NavigationLink.propTypes = {
  isActive: PropTypes.bool,
  label: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default NavigationLink;
