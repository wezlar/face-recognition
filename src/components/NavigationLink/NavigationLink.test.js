// __tests__/fetch.js
import React from 'react';
import {
  render,
  cleanup,
} from '@testing-library/react';
// this adds custom jest matchers from jest-dom
import 'jest-dom/extend-expect';
import renderWithRouter from '../../testUtils/renderWithRouter';
import NavigationLink from './NavigationLink';

// automatically unmount and cleanup DOM after the test is finished.
afterEach(cleanup);
describe(`Test NavigationLink`, () => {
  test(`Test snapshot`, () => {
    // Arrange
    const { container, getByText } = renderWithRouter(
      <NavigationLink label='Test label' url={'/home'} />,
    );
  
    // snapshots work great with regular DOM nodes!
    expect(container.firstChild).toMatchSnapshot();
  });
});