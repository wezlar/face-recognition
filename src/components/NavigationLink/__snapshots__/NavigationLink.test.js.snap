// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test NavigationLink Test snapshot 1`] = `
<li>
  <a
    class="css-1eswy7-NavigationLink"
    href="/home"
  >
    Test label
  </a>
</li>
`;
